diff -Naur fsl-6.0.4.orig/src/fdt/CUDA/diffmodels.cuh fsl-6.0.4/src/fdt/CUDA/diffmodels.cuh
--- fsl-6.0.4.orig/src/fdt/CUDA/diffmodels.cuh	2020-07-25 16:36:57.000000000 +0200
+++ fsl-6.0.4/src/fdt/CUDA/diffmodels.cuh	2020-11-15 20:35:53.824180122 +0100
@@ -66,7 +66,7 @@
     University, to negotiate a licence. Contact details are:
     fsl@innovation.ox.ac.uk quoting Reference Project 9564, FSL.*/
 
-#include <device_vector.h>
+#include <thrust/device_vector.h>
 
 void fit_PVM_single(	//INPUT
 			const vector<ColumnVector> 	datam_vec, 
diff -Naur fsl-6.0.4.orig/src/fdt/CUDA/runmcmc.cu fsl-6.0.4/src/fdt/CUDA/runmcmc.cu
--- fsl-6.0.4.orig/src/fdt/CUDA/runmcmc.cu	2020-07-25 16:36:58.000000000 +0200
+++ fsl-6.0.4/src/fdt/CUDA/runmcmc.cu	2020-11-15 20:35:53.828180122 +0100
@@ -71,8 +71,8 @@
 #include "runmcmc_kernels.cu"
 #include "sync_check.h"
 
-#include <host_vector.h>
-#include <device_vector.h> 
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h> 
 
 #include <time.h>
 #include <sys/time.h>
diff -Naur fsl-6.0.4.orig/src/fdt/CUDA/xfibres_gpu.cu fsl-6.0.4/src/fdt/CUDA/xfibres_gpu.cu
--- fsl-6.0.4.orig/src/fdt/CUDA/xfibres_gpu.cu	2020-07-25 16:36:59.000000000 +0200
+++ fsl-6.0.4/src/fdt/CUDA/xfibres_gpu.cu	2020-11-15 20:35:53.832180122 +0100
@@ -76,8 +76,8 @@
 #include "samples.h"
 #include "options.h"
 
-#include <host_vector.h>
-#include <device_vector.h> 
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h> 
 
 #include <time.h>
 #include <sys/time.h>
diff -Naur fsl-6.0.4.orig/src/fdt/CUDA/xfibres_gpu.cuh fsl-6.0.4/src/fdt/CUDA/xfibres_gpu.cuh
--- fsl-6.0.4.orig/src/fdt/CUDA/xfibres_gpu.cuh	2020-07-25 16:36:59.000000000 +0200
+++ fsl-6.0.4/src/fdt/CUDA/xfibres_gpu.cuh	2020-11-15 20:35:53.832180122 +0100
@@ -67,8 +67,8 @@
     fsl@innovation.ox.ac.uk quoting Reference Project 9564, FSL.*/
 
 #include "newimage/newimageall.h"
-#include <host_vector.h>
-#include <device_vector.h> 
+#include <thrust/host_vector.h>
+#include <thrust/device_vector.h> 
 
 #include "fibre_gpu.h"
 #include <curand_kernel.h>
diff -Naur fsl-6.0.4.orig/src/fdt/Makefile fsl-6.0.4/src/fdt/Makefile
--- fsl-6.0.4.orig/src/fdt/Makefile	2020-07-22 13:50:23.000000000 +0200
+++ fsl-6.0.4/src/fdt/Makefile	2020-11-15 20:35:53.840180122 +0100
@@ -171,7 +171,7 @@
 		   ${CXX} ${CXXFLAGS} ${LDFLAGS} -o $@ ${EDDYCOMBINEOBJS} ${DLIBS}
 
 ${LIBBEDPOSTX_CUDA}:
-		${NVCC} --shared --compiler-options '-fPIC' -o CUDA/libbedpostx_cuda.so CUDA/init_gpu.cu CUDA/samples.cu CUDA/diffmodels.cu CUDA/runmcmc.cu  CUDA/xfibres_gpu.cu -O3 ${GENCODE_FLAGS} -lcudart -lcuda -lcurand -I. -L${LIB_CUDA} -L${LIB_CUDA}/stubs -ICUDA/options -I${INC_NEWMAT} -I${FSLDIR}/include -I${INC_BOOST} -I${INC_CUDA} -I${INC_CUDA}/thrust -maxrregcount=64
+		${NVCC} --shared --compiler-options '-fPIC' -o CUDA/libbedpostx_cuda.so CUDA/init_gpu.cu CUDA/samples.cu CUDA/diffmodels.cu CUDA/runmcmc.cu  CUDA/xfibres_gpu.cu -O3 ${GENCODE_FLAGS} -lcudart -lcuda -lcurand -I. -L${LIB_CUDA} -L${LIB_CUDA}/stubs -ICUDA/options -I${INC_NEWMAT} -I${FSLDIR}/include -I${INC_BOOST} -I${INC_CUDA} -maxrregcount=64
 		@if [ ! -d ${FSLDEVDIR}/lib/ ] ; then ${MKDIR} ${FSLDEVDIR}/lib ; fi
 		${CP} -rf CUDA/libbedpostx_cuda.so ${FSLDEVDIR}/lib
 
